### PR TITLE
docker compose non hostnetwork setup

### LINKED ISSUE #2613: Docker Compose support
#### What would you like to do?

I would like to run docker compose up, with a simple YAML configuration

#### Why would you like to do it?

To avoid manual typing and tutorial hell

### LINKED ISSUE #2699: Documentation is broken
### Steps to reproduce

1. Where are you starting? What can you see?
The documentation is not in a basic state to know what is needed and how it is required to be stringed together.
2. What do you click?
https://github.com/element-hq/element-call < scroll to the readme.md
3. More steps…
read?

#### What did you expect?
At a minimum steps to get a working environment from the docs following steps. (not asking a line by line copy/past)
#### What happened instead?
well I get 50* or 404s and I happen to have to dig through docker composer files, which does not tell me if I did it right,

The files use static keys which I assume is a be idea in prod but you know it is what it is.

### Operating system

Linux

### Browser information

curl, chrome, firefox, Browsh

### URL for webapp

_No response_

### Will you send logs?

Yes

### PR DESCRIPTION
**Pull request by fkwp**
Purpose of this refactoring is to create a self containing development environment for Element Call using the localhost domain namespace:
- Minimum Synapse Setup (servername: synapse.localhost)
- LiveKit JWT Service (Note requires Federation API and hence a TLS reverse proxy)
- Minimum TLS reverse proxy (servername: synapse.localhost) Note certificates are valid for at least 10 years from now
- Minimum LiveKit SFU Setup using dev defaults for config
- Redis db for completness 

fixes https://github.com/element-hq/element-call/issues/2699
fixes https://github.com/element-hq/element-call/issues/2613

### CODE REVIEW

File: dev-backend-docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    # network rather than opening specific ports (but why?? we're not missing
+    # any…)
+    ports:
+      # HOST_PORT:CONTAINER_PORT
+      - 7880:7880/tcp
+      - 7881:7881/tcp
+      - 7882:7882/tcp
+      - 50100-50200:50100-50200/udp
+    volumes:
+      - ./backend/dev_livekit.yaml:/etc/livekit.yaml
```
Comment (AndrewFerr) [inline dev-backend-docker-compose.yml]: For compatibility with SELinux-enabled hosts, put the `:Z` flag on each volume entry:
```suggestion
      - ./backend/dev_livekit.yaml:/etc/livekit.yaml:Z
```
(and all the others)

It should be safe to use this flag even on non-SELinux systems, as documentation says it gets ignored then: https://docs.docker.com/reference/compose-file/services/#short-syntax-5

File: README.md [hunk 2]
```diff
@@ -120,25 +182,25 @@ yarn dev
...
 ### Backend
 
-A docker compose file is provided to start a LiveKit server and auth
-service for development. These use a test 'secret' published in this
-repository, so this must be used only for local development and
-**_never be exposed to the public Internet._**
+A docker compose file `dev-backend-docker-compose.yml` is provided to start the
+whole stack of components which is required for a local development environment:
 
-To use it, add a SFU parameter in your local config `./public/config.json`:
```
Comment (Half-Shot) [inline README.md:128]: We've lost the advice about copying the config over?

File: config/config.sample.json [hunk 3]
```diff
@@ -1,13 +1,10 @@
 {
   "default_server_config": {
     "m.homeserver": {
-      "base_url": "https://call.ems.host",
-      "server_name": "call.ems.host"
+      "base_url": "http://synapse.localhost:8008",
```
Comment (Half-Shot) [inline config/config.sample.json:4]: Is this change to the sample expected? Not just to the devenv.json?

**Review by Half-Shot — CHANGES_REQUESTED** [review]
I had a go with this and it worked wonderfully, just a couple of questions before approval.

File: dev-backend-docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      - ecbackend
+
+  redis:
+    image: redis:6-alpine
+    command: redis-server /etc/redis.conf
+    ports:
+      # HOST_PORT:CONTAINER_PORT
+      - 6379:6379
+    volumes:
+      - ./backend/redis.conf:/etc/redis.conf
```
Comment (AndrewFerr) [inline dev-backend-docker-compose.yml]:
```suggestion
      - ./backend/redis.conf:/etc/redis.conf:Z
```

File: dev-backend-docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      - SYNAPSE_CONFIG_PATH=/data/cfg/homeserver.yaml
+      # Needed for rootless podman-compose such that the uid/gid mapping does
+      # fit local user uid. If the container runs as root (uid 0) it is fine as
+      # it actually maps to your non-root user on the host (e.g. 1000).
+      # Otherwise uid mapping will not match your non-root user.
+      - UID=0
+      - GID=0
+    volumes:
+      - ./backend/synapse_tmp:/data
+      - ./backend/dev_homeserver.yaml:/data/cfg/homeserver.yaml
```
Comment (AndrewFerr) [inline dev-backend-docker-compose.yml]:
```suggestion
      - ./backend/synapse_tmp:/data:Z
      - ./backend/dev_homeserver.yaml:/data/cfg/homeserver.yaml:Z
```

File: dev-backend-docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      - ecbackend
+
+  nginx:
+    #  openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout tls_localhost_key.pem -out tls_localhost_cert.pem -subj "/C=GB/ST=London/L=London/O=Alros/OU=IT Department/CN=localhost"
+    hostname: synapse.localhost
+    image: nginx:latest
+    volumes:
+      - ./backend/tls_localhost_nginx.conf:/etc/nginx/conf.d/default.conf
+      - ./backend/tls_localhost_key.pem:/root/ssl/key.pem
+      - ./backend/tls_localhost_cert.pem:/root/ssl/cert.pem
```
Comment (AndrewFerr) [inline dev-backend-docker-compose.yml]:
```suggestion
      - ./backend/tls_localhost_nginx.conf:/etc/nginx/conf.d/default.conf:Z
      - ./backend/tls_localhost_key.pem:/root/ssl/key.pem:Z
      - ./backend/tls_localhost_cert.pem:/root/ssl/cert.pem:Z
```

**Review by AndrewFerr — CHANGES_REQUESTED** [review]

**Review by AndrewFerr — APPROVED** [review]

**Review by Half-Shot — APPROVED** [review]

**Pull request merged**